packages/components/src/site-thumbnail/index.tsx

```diff
@@ -23,27 +23,26 @@ export const SiteThumbnail = ( {
 	backgroundColor,
 	children,
 	alt,
-	mShotsUrl,
+	mShotsUrl = '',
 	size = 'small',
 	mshotsOption = MSHOTS_OPTION,
 }: Props ) => {
-	const maybeImage = useMshotsImg( mShotsUrl ?? '', mshotsOption );
+	const { src, isLoading, imgRef } = useMshotsImg( mShotsUrl, mshotsOption );
 
-	const src = maybeImage?.src || '';
-	const visible = !! src && mShotsUrl;
-	const textColor = backgroundColor && getTextColorFromBackground( backgroundColor );
+	const color = backgroundColor && getTextColorFromBackground( backgroundColor );
 
 	const className = classnames(
 		'site-thumbnail',
-		visible ? 'site-thumbnail-visible' : '',
+		! isLoading ? 'site-thumbnail-visible' : '',
 		`site-thumbnail__size-${ size }`
 	);
 
 	const loader = mShotsUrl ? 'site-thumbnail-loader' : '';
 
 	return (
-		<div className={ className } style={ { backgroundColor, color: textColor } }>
-			{ ! visible ? <div className={ loader }>{ children }</div> : <img src={ src } alt={ alt } /> }
+		<div className={ className } style={ { backgroundColor, color } }>
+			{ isLoading && <div className={ loader }>{ children }</div> }
+			{ src && <img ref={ imgRef } src={ src } alt={ alt } loading="lazy" /> }
```

Comment:
I noticed here that the source is always set, it's set to the mshots url and not necessarily the produced image url, this results in the unresolved image icon being displayed when mshots is loading. I would want to remove that and show only the loader. The loader is shown over that default icon.

![image](https://user-images.githubusercontent.com/47489215/181097684-f641c7c8-f100-49af-b976-4ee846c12cc9.png)

Reply:
Yep, the src is always set if it exists. It triggers the image to load and normal behaviour for the img element.
Is this appearing just a few seconds before loading the image? We could tweak some css or delay to hide that, but I could not reproduce it, even using slow 3G.